youtube_dl/extractor/camwhoresbay.py

```diff
@@ -0,0 +1,141 @@
+                    'height': int(flashvars[k.replace('_url', '_height')])
+                }
+                thumbnails.append(t)
+        return thumbnails
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._html_search_regex(r'<div class=\"headline\">\n\s*<h1>(.+?)</h1>', webpage, 'title')
+
+        flashvars = self._parse_json(
+            self._search_regex(r'var\s+flashvars\s+=\s+({\n.+});', webpage, 'flashvars', default='{}'),
+            video_id, transform_source=js_to_json)
+        uploader = self._search_regex(r'class=\"avatar\" href=\"http.+/\" title=\"(.+)\"', webpage, 'uploader',
+                                      fatal=False)
+
+        formats = self._parse_formats(flashvars)
+        thumbnails = self._parse_thumbnails(flashvars)
+
+        return {
+            'id': video_id,
+            'title': title,
+            'uploader': uploader,
+            'url': formats[0].get('url'),
+            'formats': formats,
+            'thumbnails': thumbnails,
+        }
+
+
+class CamWhoresBayModelIE(CamWhoresBayVideoIE):
+    IE_NAME = 'camwhoresbay:model'
+    _VALID_URL = r'https?://(?:www\.)?camwhoresbay\.com/models/(?P<id>[0-9a-z\-]+)/?'
+    _MORE_PAGES_INDICATOR = r'<li\s+class="next"><a\s+href="#videos"'
+    _TITLE = None
+    _TITLE_RE = r'<div\s+class=\"headline\">\n\s*<h1>\s*(.*?)\'s New Videos.+</h1>'
+    _BASE_URL_TEMPL = 'https://www.camwhoresbay.com/models/%s/?mode=async&function=get_block&block_id=list_videos_common_videos_list&sort_by=post_date&from='
+    _TEST = {
+        'url': 'https://www.camwhoresbay.com/models/kirsten-xxx/',
+        'info_dict': {
+            'id': 'kirsten-xxx',
+            'title': 'Kirsten_xxx',
+        }
+    }
+    _PAGE_SIZE = 35
+
+    def _extract_list_title(self, webpage):
+        return self._TITLE or self._html_search_regex(
+            self._TITLE_RE, webpage, 'list title', fatal=False)
+
+    def _title_and_entries(self, model_id, base_url):
+        for page in itertools.count(1):
+            webpage = self._download_webpage('%s%d' % (base_url, page), model_id, 'Downloading page %s' % page)
+
+            if page == 1:
+                yield self._extract_list_title(webpage)
+
+            for video_id in re.findall(r'<div\s+class=\"video-item\s*\">\n\s*<a href=\"(.*?)\"\s+title', webpage):
+                yield self.url_result(video_id)
+
+            if re.search(self._MORE_PAGES_INDICATOR, webpage, re.DOTALL) is None:
+                break
+
+    def _extract_videos(self, list_id, base_url):
+        title_and_entries = self._title_and_entries(list_id, base_url)
+        list_title = next(title_and_entries)
+        return self.playlist_result(title_and_entries, list_id, list_title)
+
+    def _real_extract(self, url):
+        model_id = self._match_id(url)
+        return self._extract_videos(model_id, self._BASE_URL_TEMPL % model_id)
```

Comment:
Good practice since the `%` operator can be ambiguous:
```suggestion
        return self._extract_videos(model_id, self._BASE_URL_TEMPL % (model_id, ))
```